hermes-frontend/src/main/java/pl/allegro/tech/hermes/frontend/publishing/handlers/end/TrackingHeadersExtractor.java

```diff
@@ -0,0 +1,11 @@
+package pl.allegro.tech.hermes.frontend.publishing.handlers.end;
+
+import io.undertow.util.HeaderMap;
+import java.util.Map;
+import pl.allegro.tech.hermes.frontend.publishing.metadata.HeadersToMapTransformer;
+
+public interface TrackingHeadersExtractor {
+    default Map<String, String> extractHeadersToLog(HeaderMap headers) {
+        return HeadersToMapTransformer.toHeadersMap(headers);
```

Comment:
In fact, we probably do not need this default implementation as we already have `DefaultTrackingHeaderExtractor`. Without this implementation we could remove `HeadersToMapTransformer` and not change `MessageFactory`. 

Nonetheless, I'm ok with the current approach, so I leave the decision up to you.

Reply:
you're right, done